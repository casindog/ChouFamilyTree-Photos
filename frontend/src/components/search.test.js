import { shallow } from "enzyme";
import React from 'react';
import { Search } from './search.component'

describe('<Search />', () => {
    // snapshot testing
    it('expect to render Search component', () => {
        expect(shallow(<Search />)).toMatchSnapshot()
    })

    it('correctly updates first name', () => {
        const wrapper = shallow(<Search />);
        expect(wrapper.find('input').at(0).prop('value')).toEqual('')
        wrapper.find('input').at(0).simulate('change', { target: {name: 'fname', value: 'Zoe'}})
        expect(wrapper.find('input').at(0).prop('value')).toEqual('Zoe')
    })

    it('correctly updates last name', () => {
        const wrapper = shallow(<Search />);
        expect(wrapper.find('input').at(1).prop('value')).toEqual('')
        wrapper.find('input').at(1).simulate('change', { target: {name: 'lname', value: 'Kimmel'}})
        expect(wrapper.find('input').at(1).prop('value')).toEqual('Kimmel')
    })

    it('correctly updates company', () => {
        const wrapper = shallow(<Search />);
        expect(wrapper.find('select').prop('value')).toEqual('')
        wrapper.find('select').simulate('change', { target: {name: 'company', value: 'Chevron'}})
        expect(wrapper.find("select").prop("value")).toEqual("Chevron");
    })
})

