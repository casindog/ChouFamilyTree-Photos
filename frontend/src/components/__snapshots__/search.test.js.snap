// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> expect to render Search component 1`] = `ShallowWrapper {}`;

exports[`expect to render Search component 1`] = `ShallowWrapper {}`;
